code-challenges.test.js

```diff
@@ -12,56 +12,231 @@
+}
+
+// console.log(testFunction(object1))
+// console.log(testFunction(object2))
+// console.log(testFunction(object3))
+
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // Pseudo code:
 
+//input: array of words
+//output: array with all words capitalized
+//iterate through an array using .map and then use .charAt to pick first letter of the word (index 0) then use .toUpperCase to output the first letter capitalized
+//combine that output with .slice(1) to add the first letter capital to return new array with all words capitalized
+
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("capArr", ()=> {
+    it("takes an array and makes all words capital", () =>{
+        const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+        expect(capArr(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+        const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+        expect (capArr(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+
+    })
+})
+
+// FAIL  ./code-challenges.test.js
+// testFunction
+//   ✓ takes an object and determine if the values are divisible by three (1 ms)
+// capArr
+//   ✕ takes an array and makes all words capital (1 ms)
+
+// ● capArr › takes an array and makes all words capital
+
+//   ReferenceError: capArr is not defined
+
 const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
 
+// const capArr = (array) => {
+//     return array.map((value) => {
+//         return value.charAt(0).toUpperCase() + value.slice(1)
+//     })
+// }
+
+// PASS  ./code-challenges.test.js
+// testFunction
+//   ✓ takes an object and determine if the values are divisible by three (1 ms)
+// capArr
+//   ✓ takes an array and makes all words capital (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.141 s
+// Ran all test suites.
+
+//Refactor:
+
+const capArr = (array) => { 
+    return array.map((value) => value.charAt(0).toUpperCase() + value.slice(1))
+}
+
+// console.log(capArr(randomNouns1))
+// console.log(capArr(randomNouns2))
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // Pseudo code:
 
+//input: a string that logs the index of the first vowel
+//output: a function 
+//iterate through a string and log index of first vowel
+//return index from a string with first vowel using the .includes method to compare with a variable containing vowels.
```

Comment:
Explanation is a little confusing here. Make sure you are explaining exactly what the logic is doing! Your pseudo for this one does not explain how you are getting the index of the first vowel -- you can explain that you made an array of vowels, and are checking each character in the string to see if it is in that array.